### ISSUE TITLE
Module in a different path than project won't work

### ISSUE THREAD

**Issue by dorkbox** [dubreuia/intellij-plugin-save-actions#13]
This is related to the fix for #1, because if you have a module whose code is in a different system path than the project -- the plugin won't work at all.

This is fixed (and tested against #1 also) by using

````java
// only process the project that is currently active
final DataContext dataContext = DataManager.getInstance().getDataContextFromFocus().getResult();
final Project project = DataKeys.PROJECT.getData(dataContext);
final PsiFile psiFile = PsiDocumentManager.getInstance(project).getPsiFile(document);
````

And removing "isActive" and "isPsiFilePhysicallyInProject".

[This commit](https://github.com/dorkbox/intellij-plugin-save-actions/commit/ae7a25e3eade57dd4f42308ee60fa65ca265126f) has the fix + the "only modify if saveAll command is issued" fix from @krasa as well. I would do a PR, but there are some other changes as well that you might not want.

**Comment by dubreuia** [comment]
Thanks @dorkbox, I'll check your commit this weekend and close the bug if OK

**Comment by dubreuia** [comment]
See PR : https://github.com/dubreuia/intellij-plugin-save-actions/pull/17, I not sure if we still need isActive or not...

**Comment by dubreuia** [comment]
Merged

**Comment by dubreuia** [comment]
I have a regression on #1, I'm reopening this issue.

**Comment by dorkbox** [comment]
Strange - the dataContext method should have prevented #1

**Comment by dubreuia** [comment]
Actually dataContext works as expected, the problem is that Intellij doesn't trigger file sync when you change between two editor instances, so that case do not work:

- Edit a file in editor1
- Edit a file in editor2
- Lose focus for Intellij
- Check dataContext return editor2 (last focus...), but the file sync will trigger for the files in editor1 two

I reverted the changes in master.

**Comment by dubreuia** [comment]
Corrected in 0.5
